tests/pytest_tests/system_tests/test_api/test_public_api_run_summary.py

```diff
@@ -0,0 +1,22 @@
+import wandb
+
+
+def test_delete_summary_metric_w_no_lazyload(user, test_settings):
+    run = wandb.init(settings=test_settings())
+    runid = run.id
+
+    metric = "test_val"
+    for i in range(10):
+        wandb.log({metric: i})
+    run.finish()
+
+    run = wandb.Api().run(f"uncategorized/{runid}")
+    try:
+        del run.summary[metric]
+        run.update()
+        # After deleting the metric, accessing it again should throw an error
+        run.summary[metric]
+    except KeyError:
+        assert True
+    else:
+        raise AssertionError("Metric was not deleted successfully.")
```

Comment:
you want to use https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-raises